lib/sikuli/common.sikuli/common.py

```diff
@@ -331,3 +339,72 @@ def il_hover(self, action_name, component, width, height, similarity=0.70, timeo
                     # Return location, screenshot, and time
                     return screenshot, current_time
         raise Exception('Cannot {action}'.format(action=action_name))
+
+    def _exists(self, component, similarity=0.70, timeout=10, wait_component=None):
+        """
+        Check the component base on the specify image, offset, and similarity.
+        @param component: The component you want to click. ex: GMAIL_REPLY.
+        @param similarity: The similarity of image. Default: 0.70.
+        @return: true/false
+        """
+        # get the loop time base on the pattern amount of component, min loop time is 10 times
+        wait_sec = 0.5
+        loop_time = self._get_loop_times(object_amount=len(component), total_second=timeout, each_check_second=wait_sec)
+        for counter in range(loop_time):
+            for pic, offset_x, offset_y in component:
+                p = Pattern(pic).similar(similarity).targetOffset(offset_x, offset_y)
```

Comment:
fulfill the pattern of the image, I will leave the interface here, so if user gives the parameter 0,0, that should not affect the result?